File: src/dataset/Java/twelve-days_exercism-shashwata27/gemini/GeminiFlash_TwelveDays_v4.java
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class TwelveDays {

    private static final String[] ORDINALS = {
            "first", "second", "third", "fourth", "fifth",
            "sixth", "seventh", "eighth", "ninth", "tenth",
            "eleventh", "twelfth"
    };

    private static final String[] GIFTS = {
            "a Partridge in a Pear Tree",
            "two Turtle Doves",
            "three French Hens",
            "four Calling Birds",
            "five Gold Rings",
            "six Geese-a-Laying",
            "seven Swans-a-Swimming",
            "eight Maids-a-Milking",
            "nine Ladies Dancing",
            "ten Lords-a-Leaping",
            "eleven Pipers Piping",
            "twelve Drummers Drumming"
    };

    String verse(int verseNumber) {
        StringBuilder verse = new StringBuilder();
        verse.append("On the ").append(ORDINALS[verseNumber - 1]).append(" day of Christmas my true love gave to me: ");

        for (int i = verseNumber - 1; i >= 0; i--) {
            verse.append(GIFTS[i]);
            if (i > 0) {
                verse.append(", ");
            } else if (verseNumber > 1) {
                verse.append(", and ");
            }
        }

        verse.append(".\n");
        return verse.toString();
    }

    String verses(int startVerse, int endVerse) {
        return IntStream.rangeClosed(startVerse, endVerse)
                .mapToObj(this::verse)
                .collect(Collectors.joining("\n"));
    }

    String sing() {
        return verses(1, 12);
    }
}

File: src/dataset/typescript/twelve-days_exercism-wackerow/gemini/GeminiFlash_twelve-days_v4.ts
class TwelveDays {
    private static readonly _numbers: string[] = [
        'first',
        'second',
        'third',
        'fourth',
        'fifth',
        'sixth',
        'seventh',
        'eighth',
        'ninth',
        'tenth',
        'eleventh',
        'twelfth',
    ];

    private static readonly _gifts: string[] = [
        'a Partridge in a Pear Tree.\n',
        'two Turtle Doves, ',
        'three French Hens, ',
        'four Calling Birds, ',
        'five Gold Rings, ',
        'six Geese-a-Laying, ',
        'seven Swans-a-Swimming, ',
        'eight Maids-a-Milking, ',
        'nine Ladies Dancing, ',
        'ten Lords-a-Leaping, ',
        'eleven Pipers Piping, ',
        'twelve Drummers Drumming, '
    ];

    public static recite(fromVerse: number, toVerse: number): string {
        let song = '';
        for (let i = fromVerse - 1; i < toVerse; i++) {
            let verse = `On the ${TwelveDays._numbers[i]} day of Christmas my true love gave to me: `;
            let gifts = '';
            for (let j = i; j >= 0; j--) {
                gifts += (i > 0 && j === 0) ? 'and ' + TwelveDays._gifts[0] : TwelveDays._gifts[j];
            }
            verse += gifts;
            song += verse;
        }
        return song;
    }
}

export default TwelveDays